### DISCUSSION TITLE
Improve "List jobs for a workflow run" data

### DISCUSSION THREAD

**Discussion by patricioperezv** [community/community#83515] [Enterprise]
### Select Topic Area

Product Feedback

### Body

Hey there, i've been using actions for a while and i would like to gather exciting data from the github api about the usage of steps inside a workflow job.
While some data is exposed in the api about each one of the job's steps i see room for improvement, what i would like to see is the following data:
* action name/version (Can be grabbed from the uses section in the workflow step definition)
* action parameters (Can be grabbed from the with section in the workflow step definition)
* shell used if this is a run step
* run command if this is a run step

I've tried some workarounds like grabbing the workflow definition for a specific workflow run, but i don't see a good way to correlate data correctly (Mainly there's no 1:1 relationship between all the steps in the workflow definition and the actual jobs, i.e there are more steps in the actual execution, like building containers for certain actions). And also, going with the idea of just checking the workflow definition, there's no way to actually access variables that can be used in the data points i need (i.e if some user go and use actions/checkout@${{ vars.MY_ACTIONS_CHECKOUT_VERSION }} i can't actually know what was the value for that variable in a clean way). Another idea was to use the data from the name field, which if the user doesn't override the name in the workflow step, it shows the used action/version, but yeah, it's bad when users override it.
Another problem with this approach is checking if the step actually ran (Maybe the previous one failed, or it has an if conditional that skipped it based on the condition.
There's also a way of grabbing that data from the logs, but that would require log parsing and seems a little bit of an overkill to grab that data.

TLDR: Please add action name/version and action parameters to the workflow jobs definitions of the api, this will help us to generate data about the usage of actions/steps over a large repo base and improve the way we leverage actions on our enterprise.
